platform_umbrella/apps/kube_services/lib/kube_services/stale/stale.ex

```diff
@@ -26,6 +28,7 @@ defmodule KubeServices.Stale do
 
   def stale?(resource, seen_res_set \\ nil)
   def stale?(resource, nil), do: stale?(resource, recent_resource_map_set())
+  def stale?(resource, seen_res_set) when seen_res_set == @empty, do: stale?(resource, recent_resource_map_set())
```

Comment:
Is there a concern here of infinite regression in case the recent resources is persistently empty? Seems unlikely...

Reply:
I think you're right, and it's safe.